### PR TITLE
report cbde build failures

### PR DESCRIPTION
**Pull request by geoffray-adde-sonarsource**
un peu plus que 2 lignes au final :-/

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs [hunk 1]
```diff
@@ -183,6 +184,32 @@ private static Diagnostic CreateDiagnosticFromJToken(JToken token)
...
+            var begin = new LinePosition(1,1);
+            var end = new LinePosition(1,2);
+            var loc = Location.Create(filename, TextSpan.FromBounds(0, 0), new LinePositionSpan(begin, end));
+
+            return Diagnostic.Create(rule, loc);
+        }
+        private void RaiseIssueFromFailedCbdeRun(CompilationAnalysisContext context)
+        {
+            string failureString = "CBDE Failure Report :\n  C# souces files involved are:\n";
+            foreach (var fileName in csSourceFileNames)
```
Comment (loic-joly-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs]: Use `string.join`

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs [hunk 2]
```diff
@@ -183,6 +184,32 @@ private static Diagnostic CreateDiagnosticFromJToken(JToken token)
...
+
+            var begin = new LinePosition(1,1);
+            var end = new LinePosition(1,2);
+            var loc = Location.Create(filename, TextSpan.FromBounds(0, 0), new LinePositionSpan(begin, end));
+
+            return Diagnostic.Create(rule, loc);
+        }
+        private void RaiseIssueFromFailedCbdeRun(CompilationAnalysisContext context)
+        {
+            string failureString = "CBDE Failure Report :\n  C# souces files involved are:\n";
```
Comment (loic-joly-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs]: Use a `StringBuilder`

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs (hunk 1 above)
Comment (geoffray-adde-sonarsource) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs]: I am just using the StringBuilder as suggested below as re-configuring the code to use string.Join would be far less readable for a negligible perf boost.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs (hunk 2 above)
Comment (geoffray-adde-sonarsource) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs]: done

**Pull request merged**